src/drivers/intel/cavs/dmic.c

```diff
@@ -383,14 +383,11 @@ static void find_modes(struct decim_modes *modes, uint32_t fs, int di)
 		 * is possible. Then check that CIC decimation constraints
 		 * are met. The passed decimation modes are added to array.
 		 */
-		j = 0;
-		/* Loop until NULL */
-		while (fir_list[j]) {
+		for (j = 0; fir_list[j]; j++) {
```

Comment:
I suppose this works but what is the improvement?

Reply:
I find
```
for (i = 0; i < N; i++)
```
better than
```
i = 0;
while (i < N) {
...
i++;
}
```